src/System.Collections/src/System/Collections/Generic/LinkedList.cs

```diff
@@ -188,7 +188,7 @@ public void CopyTo(T[] array, int index)
 
             if (index < 0 || index > array.Length)
```

Comment:
There's a micro-opt possible with this patter where one uses `if ((uint)index > (uint)array.Length)` or equivalent. While at the C# level that's got two casts, it compiles to a `cgt.un` rather than any explicit caching operation, so the effect is one less branch and one less comparison in the most common case.
Whether that's worth doing or not is another question.